lisa/util/process.py

```diff
@@ -180,17 +189,23 @@ def _process_command(
                 split_command = ["cmd", "/c", command]
             else:
                 split_command = []
+                if nohup:
+                    split_command += ["nohup"]
                 if sudo:
                     split_command += ["sudo"]
                 envs = _create_exports(update_envs=update_envs)
                 if envs:
                     command = f"{envs} {command}"
+
                 split_command += ["sh", "-c", command]
                 # expand variables in posix mode
                 update_envs.clear()
         else:
-            if sudo and self._is_posix:
-                command = f"sudo {command}"
+            if self._is_posix:
+                if sudo:
+                    command = f"sudo {command}"
+                if nohup:
+                    command = f"nohup {command}"
```

Comment:
Does it support kill now? If yes, it can be enabled by default without an argument. And is it tested to work with sudo and shell well?

Reply:
We should test on a wide set of OS's, I have not yet so this PR is just for review at the moment. 
The full documentation notes some shells implement nohup as a builtin, so it's possible (like kill) that there are multiple paths to call it if the gnucoreutil binary and the shell are both available. That implies it may also be possible that the binary is not present and only the shell builtin is present, which would break this logic.

My understanding was that you should always call nohup first if you're using it.

Reply:
> Does it support kill now? 
It should, nohup only catches SIGHUP not SIGKILL etc

Reply:
re testing: this PR needs substantial testing across OS's. It's possible nohup is implemented as a shell builtin and not a binary on some systems, this will break the current logic since nohup is always called first to ensure the calling process isn't killed on disconnect.

> Does it support kill now? And is it tested to work with sudo and shell well?

The full documentation notes some shells implement nohup as a builtin, so it's possible (like kill) that there are multiple paths to call it if the gnucoreutil binary and the shell are both available. That implies it may also be possible that the binary is not present and only the shell builtin is present, which would break this logic.

>If yes, it can be enabled by default without an argument. 

No, nohup redirects stdin and stdout to a file so it is not good for any process except on where you don't care about the output as much. Callers will have to collect the nohup.out file to get their stdout after reconnecting if they need it.